Plugins/SwiftProtobufPlugin/plugin.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+import PackagePlugin
+
+@main
+struct SwiftProtobufPlugin: BuildToolPlugin {
+    /// Errors thrown by the `SwiftProtobufPlugin`
+    enum PluginError: Error {
+        /// Indicates that the target where the plugin was applied to was not `SourceModuleTarget`.
+        case invalidTarget
+        /// Indicates that no config file was present in the root of the target directory.
+        case noConfigFile
+    }
+
+    /// The configuration of the plugin.
+    struct Configuration: Codable {
+        /// Encapsulates a single invocation of protoc.
+        struct Invocation: Codable {
+            /// The visibility of the generated files.
+            enum Visibility: String, Codable {
+                /// The generated files should have `internal` access level.
+                case `internal`
+                /// The generated files should have `public` access level.
+                case `public`
+            }
+
+            /// An array of paths to `.proto` files for this invocation.
+            var protoFiles: [String]
+            /// The visibility of the generated files.
+            var visibility: Visibility?
+        }
+
+        /// The path to the `protoc` binary.
+        ///
+        /// If this is not set, SPM will try to find the tool itself.
+        var protocPath: String?
+
+        /// A list of invocations of `protoc` with the `SwiftProtobuf` plugin.
+        var invocations: [Invocation]
+    }
+
+    static let configurationFileName = "swift-protobuf-config.json"
+
+    func createBuildCommands(context: PluginContext, target: Target) throws -> [Command] {
+        print("Creating build commands (\(UUID()))")
```

Comment:
Is this useful?  most of the time it seems like having builds be complete silent is what folks prefer.

Reply:
I removed this again. This was very useful while developing the plugin because SPM had some hiccups during the 5.7 Betas. Though all of them are fixed up now.